### PR TITLE
Handlebars Templates for Events

### PR DESCRIPTION
**Pull request by joshgummersall**
Adds support for handlebars based templates for event alerts.

Fixes https://github.com/joshgummersall/statsd-alerting-backend/issues/10

### CODE REVIEW

**Comment by joshgummersall** [comment]
@cwhitten you think you'll have a chance to review this change?

File: src/alerts/hipchat.coffee [hunk 1]
```diff
@@ -3,6 +3,8 @@ Hipchatter = require 'hipchatter'
 
 module.exports = class HipChatAlert extends Alert
   constructor: (@config) ->
+    super @config
```
Comment (cwhitten) [inline src/alerts/hipchat.coffee]: I don't think you have to pass `@config` as a parameter in a `super` call

File: src/alerts/hipchat.coffee [hunk 2]
```diff
@@ -11,26 +13,31 @@ module.exports = class HipChatAlert extends Alert
...
-    @api.notify @config.room, options, (err, response) ->
-      if err
-        throw new Error "[Hipchat] Couldn't send message to room: #{err}"
+    }
+    @api.notify @config.room, options, (err) ->
+      throw new Error "[Hipchat] Couldn't send message to room: #{err}" if err
 
   sendEvent: (event) ->
-    @sendToHipChat [
+    eventString = @renderEvent event
```
Comment (cwhitten) [inline src/alerts/hipchat.coffee]: This pattern of string assignment could be DRY'd up a bit by moving the default format into the Alerts class and doing a:

```coffeescript
eventString = @renderEvent(event) or @defaultEvent(event)
```

Same with the metrics string assignment

```coffeescript
eventString = @renderEvent(event) or @defaultMetricsEvent(event)
```

**Comment by cwhitten** [comment]
ping

**Comment by joshgummersall** [comment]
@cwhitten I hit your comments, mind taking another look? Thanks.

**Comment by cwhitten** [comment]
Nice. LGTM!

**Comment by joshgummersall** [comment]
@chook can you take a look at the changes here (the README changes describe the templating functionality) and see what you think? I plan to get this merged and released today or tomorrow.

**Comment by chook** [comment]
@joshgummersall i can only do it tomorrow morning (GMT+3)..

**Comment by joshgummersall** [comment]
@chook I wanted to get a chance to test things end to end and just did - tested the template functionality and the Slack webhook fix you submitted. I'll merge this up and publish it, please leave any issues you run into.

**Pull request merged**